articles/quickstart/backend/golang/interactive.md

```diff
@@ -0,0 +1,116 @@
+
+Alternatively, you can read [our getting started guide](get-started/auth0-overview/set-up-apis) that helps you set up your first API through the Auth0 dashboard.
+
+Every API in Auth0 is configured using an API Identifier that your application code will use as the Audience to validate the access token.
+
+<%= include('../../../_includes/_api_auth_intro') %>
+
+## Define permissions
+<%= include('../_includes/_api_scopes_access_resources') %>
+
+## Install dependencies
+
+Add a `go.mod` file to list all the dependencies to be used.
+
+```text
+// go.mod
+
+module 01-Authorization-RS256
+
+go 1.16
+
+require (
+	github.com/auth0/go-jwt-middleware/v2 v2.0.0
+	github.com/joho/godotenv v1.4.0
+)
+```
+
+Download dependencies by running the following shell command:
+
+```shell
+go mod download
+```
+
+## Configure your application
+
+Create a `.env` file within the root of your project directory to store the app configuration, and fill in the
+environment variables:
+
+```sh
+# The URL of our Auth0 Tenant Domain.
+# If you're using a Custom Domain, be sure to set this to that value instead.
+AUTH0_DOMAIN='${account.namespace}'
+
+# Our Auth0 API's Identifier.
+AUTH0_AUDIENCE='${apiIdentifier}'
+```
+
+## Create a middleware to validate access tokens {{{ data-action=code data-code="middleware/jwt.go" }}}
+
+Access token validation will be done in the `EnsureValidToken` middleware function which can be applied to any
+endpoints you wish to protect.
+
+If the token is valid, the resources which are served by the endpoint can be released,
+otherwise a `401 Authorization` error will be returned.
+
+Setup the **go-jwt-middleware** middleware to verify access tokens from incoming requests.
+
+<%= include('../_includes/_api_jwks_description_no_link') %>
+
+In addition to this, include a mechanism for checking that the token has the sufficient **scope** to access the requested resources.
+
+Create a function `HasScope` to check and ensure the access token has the correct scope before returning a successful response.
+
+## Protect API endpoints {{{ data-action=code data-code="main.go" }}}
+
+In this example, create an `/api/public` endpoint that does not use the `EnsureToken` middleware as it is accessible to non-authenticated requests.
+
+Create an `/api/private` endpoint that requires the `EnsureToken` middleware as it is only available to authenticated requests containing an access token with no additional scope.
+
+Create an `/api/private-scoped` endpoint that requires the `EnsureToken` middleware and `HasScope` as it is only available for authenticated requests containing an access token with the `read:messages` scope granted.
+
+::: note
+Only the `read:messages` scope is checked. You may want to extend the `HasScope` function or make it a standalone middleware that accepts multiple scopes to fit your use case.
```

Comment:
checked by what?

Reply:
@adamjmcgrath could you clarify this? I'm assuming it means that only the `read:messages` scope is validated "by the API"

Reply:
How about?

```suggestion
Only the `read:messages` scope is checked by the `HasScope` function, you may want to extend it or make it a standalone middleware that accepts multiple scopes to fit your use case.
```

Reply:
```suggestion
The `HasScope` function is the only check on the `read:messages` scope. You may want to extend the scope or make a standalone middleware to accept multiple scopes.
```

Still feels kind of awkward but if you think one is clearer, feel free to commit and merge.